File: client/verify_lab.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();

  try {
    console.log('Navigating to Lab...');
    await page.goto('http://localhost:5173/lab');

    // Wait for the page to load
    await page.waitForSelector('.lab-page');

    console.log('Lab page loaded.');

    // Find the HCl slider
    // I added id="hcl-input"
    const hclInput = page.locator('#hcl-input');

    // Check initial color of test tube
    // In CustomTestTube: <path id="permanent" fill={color} ... />
    const tubeLiquid = page.locator('#permanent');

    // Drag slider to 50
    console.log('Changing HCl input...');
    await hclInput.evaluate(e => {
      e.value = 50;
      e.dispatchEvent(new Event('change', { bubbles: true }));
      // Also trigger input just in case, though the code uses onChange
      e.dispatchEvent(new Event('input', { bubbles: true }));
    });

    // Wait for update
    await page.waitForTimeout(1000);

    // Check new color
    const newFill = await tubeLiquid.getAttribute('fill');
    console.log('New fill:', newFill);

    // Take screenshot
    await page.screenshot({ path: 'verification_lab.png' });
    console.log('Screenshot saved to verification_lab.png');

  } catch (e) {
    console.error(e);
  } finally {
    await browser.close();
  }
})();

File: verify_lab.py
from playwright.sync_api import sync_playwright

def run(playwright):
    browser = playwright.chromium.launch()
    page = browser.new_page()

    try:
        # Navigate
        print("Navigating to http://localhost:5173/lab")
        page.goto("http://localhost:5173/lab")

        # Wait for page to load
        page.wait_for_selector(".lab-page", timeout=10000)
        print("Page loaded")

        # Find HCl slider
        hcl_input = page.locator("#hcl-input")

        # Check initial color
        tube_liquid = page.locator("#permanent")
        print(f"Initial fill: {tube_liquid.get_attribute('fill')}")

        # Use keyboard to change value
        print("Changing HCl input using keyboard...")
        hcl_input.focus()
        # Press right arrow 10 times
        for _ in range(10):
            page.keyboard.press("ArrowRight")

        # Wait for update
        page.wait_for_timeout(1000)

        # Check new color
        new_fill = tube_liquid.get_attribute("fill")
        print(f"New fill: {new_fill}")

        # Also check the input value
        input_val = hcl_input.input_value()
        print(f"Input value: {input_val}")

        if new_fill == "#05B9C4":
            print("VERIFICATION SUCCESS: Color updated correctly!")
        else:
            print(f"VERIFICATION FAILURE: Expected #05B9C4, got {new_fill}")

        # Take screenshot
        page.screenshot(path="verification_lab.png")
        print("Screenshot saved to verification_lab.png")

    except Exception as e:
        print(f"Error: {e}")
    finally:
        browser.close()

with sync_playwright() as playwright:
    run(playwright)
